frontend/src/routes/password.recovery.index.lazy.tsx

```diff
@@ -0,0 +1,152 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  createLazyFileRoute,
+  useRouter,
+  useSearch,
+} from "@tanstack/react-router";
+import IconLockSolid from "@vector-im/compound-design-tokens/assets/web/icons/lock-solid";
+import { Alert, Form } from "@vector-im/compound-web";
+import { FormEvent } from "react";
+import { useTranslation } from "react-i18next";
+import { useMutation, useQuery } from "urql";
+
+import BlockList from "../components/BlockList";
+import Layout from "../components/Layout";
+import LoadingSpinner from "../components/LoadingSpinner";
+import PageHeading from "../components/PageHeading";
+import PasswordCreationDoubleInput from "../components/PasswordCreationDoubleInput";
+import { graphql } from "../gql";
+import { SetPasswordStatus } from "../gql/graphql";
+import { translateSetPasswordError } from "../i18n/password_changes";
+
+const QUERY = graphql(/* GraphQL */ `
+  query PasswordRecoveryQuery {
+    siteConfig {
+      id
+      ...PasswordCreationDoubleInput_siteConfig
+    }
+  }
+`);
+
+// TODO
+const RECOVER_PASSWORD_MUTATION = graphql(/* GraphQL */ `
+  mutation RecoverPassword($ticket: String!, $newPassword: String!) {
+    setPasswordByRecovery(
+      input: { ticket: $ticket, newPassword: $newPassword }
+    ) {
+      status
+    }
+  }
+`);
+
+export const Route = createLazyFileRoute("/password/recovery/")({
+  component: RecoverPassword,
+});
+
+function RecoverPassword(): React.ReactNode {
+  const { t } = useTranslation();
+  const { ticket } = useSearch({
+    from: "/password/recovery/",
+  });
+  const [queryResult] = useQuery({ query: QUERY });
+  const router = useRouter();
+  if (queryResult.error) throw queryResult.error;
+  const siteConfig = queryResult.data?.siteConfig;
+  if (!siteConfig) throw Error(); // This should never happen
+
+  const [result, changePassword] = useMutation(RECOVER_PASSWORD_MUTATION);
+
+  const onSubmit = async (event: FormEvent<HTMLFormElement>): Promise<void> => {
+    event.preventDefault();
+
+    const formData = new FormData(event.currentTarget);
+
+    const newPassword = formData.get("new_password") as string;
+    const newPasswordAgain = formData.get("new_password_again") as string;
+
+    if (newPassword !== newPasswordAgain) {
+      throw new Error("passwords mismatch; this should be checked by the form");
+    }
+
+    const response = await changePassword({ ticket, newPassword });
+
+    if (
+      response.data?.setPasswordByRecovery.status === SetPasswordStatus.Allowed
+    ) {
+      // TODO HOW DO WE NAVIGATE
+      router.navigate({ to: "/../login" });
```

Comment:
Honestly `window.location.href = '/login'` might be fine. Part of the annoying thing here is getting the exact app root, as MAS might be mounted on a subpath.

Maybe the best we can do is load the react app root (aka `/account`) but with a full page load, so something like

```ts
const router = useRouter(); // not in the if, because of the rules of hooks

const location = router.buildLocation({ to: "/" });
window.location.href = location.href;
```

Reply:
Yup that works